ampd/src/queue/queued_broadcaster.rs

```diff
@@ -322,31 +324,36 @@ mod test {
             .expect_broadcast()
             .once()
             .returning(move |msgs| {
-                assert!(msgs.len() == tx_count - 1);
+                assert_eq!(msgs.len(), tx_count - 1);
 
                 Ok(TxResponse::default())
             });
         broadcaster
             .expect_broadcast()
             .once()
             .returning(move |msgs| {
-                assert!(msgs.len() == 1);
-
+                assert_eq!(msgs.len(), 1);
                 Ok(TxResponse::default())
             });
 
-        let queued_broadcaster = QueuedBroadcaster::new(
-            broadcaster,
-            batch_gas_limit,
-            tx_count,
-            Duration::from_secs(5),
-        );
+        let mut desired_interval = interval(Duration::from_secs(5));
+        select! {
+            _ = desired_interval.tick() => {
+            },
+        }
+
+        let queued_broadcaster =
+            QueuedBroadcaster::new(broadcaster, batch_gas_limit, tx_count, desired_interval);
         let client = queued_broadcaster.client();
         let handle = tokio::spawn(queued_broadcaster.run());
 
         for _ in 0..tx_count {
             client.broadcast(dummy_msg()).await.unwrap();
         }
+
+        // Advance time to cause the last message broadcast
+        advance(Duration::from_secs(7)).await;
```

Comment:
can you remove this? the broadcaster should actually do a final broadcast on shutdown (after `drop(client)`), so we don't have to rely on the ticker here at all

Reply:
yeah I removed it and the test was fine.